### PR TITLE
Check Test Logs for Errors

### PR DESCRIPTION
**Pull request by kalverra**
Enables checking Chainlink logs when tests are finished for any concerning error levels, depending on what the test does.

Also preps ground work to include Pyroscope metric collection in CI

### CODE REVIEW

File: integration-tests/contracts/ethereum_vrf_contracts.go [hunk 1]
```diff
@@ -705,26 +705,37 @@ func (dkgContract *EthereumDKG) WaitForTransmittedEvent() (*dkg.DKGTransmitted,
...
 	defer subscription.Unsubscribe()
-	transmittedEvent := <-transmittedEventsChannel
-	if err != nil {
-		return nil, err
+
+	for {
+		select {
+		case err = <-subscription.Err():
+			return nil, err
+		case <-time.After(time.Minute * 5): // this can take a while
```
Comment (iljapavlovs) [inline integration-tests/contracts/ethereum_vrf_contracts.go:713]: can we provide the timeout from the test itself? rationale - to have just right amount of information available from test itself; for the test to be more descriptive

File: integration-tests/contracts/ethereum_vrf_contracts.go [hunk 2]
```diff
@@ -705,26 +705,37 @@ func (dkgContract *EthereumDKG) WaitForTransmittedEvent() (*dkg.DKGTransmitted,
...
 	defer subscription.Unsubscribe()
-	configSetEvent := <-configSetEventsChannel
-	if err != nil {
-		return nil, err
+
+	for {
+		select {
+		case err = <-subscription.Err():
+			return nil, err
+		case <-time.After(time.Minute):
```
Comment (iljapavlovs) [inline integration-tests/contracts/ethereum_vrf_contracts.go:733]: can we provide the timeout from the test itself? rationale - to have just right amount of information available from test itself; for the test to be more descriptive

File: integration-tests/smoke/vrfv2_test.go [hunk 3]
```diff
@@ -167,17 +169,17 @@ func setupVRFv2Test(t *testing.T) (testEnvironment *environment.Environment, tes
 		})
 	}
 
-	networkDetailTOML := `[EVM.GasEstimator]
-LimitDefault = 1400000
-PriceMax = 100000000000
-FeeCapDefault = 100000000000`
+	// 	networkDetailTOML := `[EVM.GasEstimator]
```
Comment (iljapavlovs) [inline integration-tests/smoke/vrfv2_test.go:172]: dont forget to uncomment

File: integration-tests/smoke/vrfv2_test.go [hunk 4]
```diff
@@ -167,17 +169,17 @@ func setupVRFv2Test(t *testing.T) (testEnvironment *environment.Environment, tes
...
+	// FeeCapDefault = 100000000000`
 	testEnvironment = environment.New(&environment.Config{
 		NamespacePrefix: fmt.Sprintf("smoke-vrfv2-%s", strings.ReplaceAll(strings.ToLower(testNetwork.Name), " ", "-")),
 		Test:            t,
 	}).
 		AddHelm(evmConfig).
-		AddHelm(chainlink.New(0, map[string]interface{}{
-			"toml": client.AddNetworkDetailedConfig("", networkDetailTOML, testNetwork),
+		AddHelm(chainlink.New(0, map[string]any{
+			"toml": client.AddNetworkDetailedConfig("", config.DefaultOCR2VRFNetworkDetailTomlConfig, testNetwork),
```
Comment (iljapavlovs) [inline integration-tests/smoke/vrfv2_test.go:182]: VRF v2 - is not OCR2VRF.
OCR2VRF is VRF v3

File: integration-tests/smoke/ocr2vrf_test.go [hunk 5]
```diff
@@ -162,7 +163,9 @@ func setupOCR2VRFEnvironment(t *testing.T) (testEnvironment *environment.Environ
 			"replicas": "6",
 			"toml": client.AddNetworkDetailedConfig(
 				config.BaseOCR2VRFTomlConfig,
-				config.DefaultOCR2VRFNetworkDetailTomlConfig, testNetwork),
+				config.DefaultOCR2VRFNetworkDetailTomlConfig,
+				testNetwork,
```
Comment (iljapavlovs) [inline integration-tests/smoke/ocr2vrf_test.go:167]: hmm, curious how it was working without providing `testNetwork`

File: integration-tests/contracts/ethereum_vrf_contracts.go [hunk 6]
```diff
@@ -905,11 +922,17 @@ func (beacon *EthereumVRFBeacon) WaitForNewTransmissionEvent() (*vrf_beacon.VRFB
...
 	defer subscription.Unsubscribe()
-	newTransmissionEvent := <-newTransmissionEventsChannel
-	if err != nil {
-		return nil, err
+
+	for {
+		select {
+		case err := <-subscription.Err():
+			return nil, err
+		case <-time.After(5 * time.Minute):
```
[the remark this answers is not in the document]
Comment (kalverra) [reply integration-tests/contracts/ethereum_vrf_contracts.go:930]: That makes sense to me, where would you prefer we provide the timeout though? We have a default-per-network, but I'm not sure that makes sense for this use-case.

**Review by tateexon — APPROVED** [review]

**Pull request merged**